src/test/test_pidfile.sh

```diff
@@ -71,7 +71,7 @@ function TEST_pidfile() {
     # another daemon
     cp $dir/osd.0.pid $dir/osd.0.pid.old  # so that kill_daemon finds the pid
```

Comment:
@dachary 
So I think we need to put osd.0.pid.old in another directory, such as $old/osd.0.pid，and
kill_daemons $old TERM osd.0 || return 1 , right?